### PR TITLE
refactor: Use `netstandard2.0` as target framework to support wider frameworks

### PR DESCRIPTION
**Pull request by nvborisenko**
## List of changes

Currently Appium targets to support:
- net6.0
- net framework 4.8
 
It means that I cannot use Appium with .net framework 4.7

## Types of changes

Use `netstandard2.0` to cover more frameworks.

https://learn.microsoft.com/en-us/dotnet/standard/net-standard?tabs=net-standard-2-0#when-to-target-net80-vs-netstandard

> For existing code that targets .NET Standard 2.0 or later, there's no need to change the TFM to net8.0 or a later TFM. The only reason to retarget from .NET Standard to .NET 8+ would be to gain access to more runtime features, language features, or APIs.

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change that adds functionality or value)
- [ ] Breaking change (fix or feature that would cause existing functionality not to work as expected)
- [ ] **Test fix** (non-breaking change that improves test stability or correctness)

## Documentation
- [ ] Have you proposed a file change/ PR with Appium to update documentation? 
## Integration tests
- [ ] Have you provided integration tests for your changes? (required for Bugfix, New feature, or Test fix)

## Details

Please provide more details about changes if necessary. You can provide code samples showing how they work and possible use cases if there are new features. Also, you can create [gists](https://gist.github.com) with pasted C# code samples or put them here using markdown. 
About markdown please read [Mastering markdown](https://guides.github.com/features/mastering-markdown/) and [Writing on GitHub](https://docs.github.com/en/get-started/writing-on-github)

### CODE REVIEW

File: src/Appium.Net/Appium.Net.csproj [hunk 1]
```diff
@@ -1,14 +1,14 @@
 <Project Sdk="Microsoft.NET.Sdk">
   <PropertyGroup>
-    <TargetFrameworks>net48;net6.0</TargetFrameworks>
-    <PackageLicenseFile>LICENSE.txt</PackageLicenseFile>
```
Comment (Dor-bl) [inline src/Appium.Net/Appium.Net.csproj:4]: Why was this removed?

File: src/Appium.Net/Appium.Net.csproj [hunk 2]
```diff
@@ -19,40 +19,25 @@
...
     <PackageLicenseFile>LICENSE.txt</PackageLicenseFile>
-  </PropertyGroup>
-  <PropertyGroup>
     <PackageReadmeFile>README.md</PackageReadmeFile>
-  </PropertyGroup>
-  <PropertyGroup>
     <GenerateDocumentationFile>true</GenerateDocumentationFile>
   </PropertyGroup>
-  <PropertyGroup>
-    <LangVersion>latest</LangVersion>
```
Comment (Dor-bl) [inline src/Appium.Net/Appium.Net.csproj:33]: Shouldn't be we keep it?

File: src/Appium.Net/Appium/Service/AppiumLocalService.cs [hunk 3]
```diff
@@ -65,24 +61,14 @@ internal AppiumLocalService(
...
             InitializationTimeout = initializationTimeout;
             EnvironmentForProcess = environmentForProcess;
-#if !NET48
-            SharedHttpClient = CreateHttpClientInstance;
-#endif
+            SharedHttpClient = CreateHttpClientInstance();
         }
 
-#if !NET48
-        private HttpClient CreateHttpClientInstance
```
Comment (Dor-bl) [inline src/Appium.Net/Appium/Service/AppiumLocalService.cs:74]: @nvborisenko have you run some tests against .NET framework 4.7+?

File: src/Appium.Net/Appium.Net.csproj (hunk 1 above)
Comment (nvborisenko) [reply src/Appium.Net/Appium.Net.csproj:4]: Removed and added `netstandard2.0`

Comment (nvborisenko) [reply src/Appium.Net/Appium.Net.csproj:4]: If you mean `licence.txt`, then it was duplicated.

File: src/Appium.Net/Appium.Net.csproj (hunk 2 above)
Comment (nvborisenko) [reply src/Appium.Net/Appium.Net.csproj:33]: Removed here, added upper.

File: src/Appium.Net/Appium/Service/AppiumLocalService.cs (hunk 3 above)
Comment (nvborisenko) [reply src/Appium.Net/Appium/Service/AppiumLocalService.cs:74]: Do you know any known issues?

Comment (nvborisenko) [reply src/Appium.Net/Appium/Service/AppiumLocalService.cs:74]: Only .net framework 4.7 is under interest? `netstandard2.0` is supported by .net framework 4.6.1

Comment (Dor-bl) [reply src/Appium.Net/Appium/Service/AppiumLocalService.cs:74]: You said you were trying to run with. Net 4.7, so this is why I asked. 
:)

Comment (Dor-bl) [reply src/Appium.Net/Appium/Service/AppiumLocalService.cs:74]: > Do you know any known issues?

Should be the same issues I had for 4.8 😁

Comment (nvborisenko) [reply src/Appium.Net/Appium/Service/AppiumLocalService.cs:74]: Really, what issues?

Comment (Dor-bl) [reply src/Appium.Net/Appium/Service/AppiumLocalService.cs:74]: Basically when I reintroduced 4.8.
https://github.com/appium/dotnet-client/pull/740
Maybe things have changed since then.
I can have a look later if you unable to run the tests

Comment (nvborisenko) [reply src/Appium.Net/Appium/Service/AppiumLocalService.cs:74]: LOL, now I see:

image

Comment (nvborisenko) [reply src/Appium.Net/Appium/Service/AppiumLocalService.cs:74]: Just believe me, this PR is about "Use netstandard2.0 as target framework to support **wider frameworks**"

Comment (nvborisenko) [reply src/Appium.Net/Appium/Service/AppiumLocalService.cs:74]: I am not from Selenium team, I am contributing as personal.

Comment (nvborisenko) [reply src/Appium.Net/Appium/Service/AppiumLocalService.cs:74]: As for Selenium:

image

Comment (Dor-bl) [reply src/Appium.Net/Appium/Service/AppiumLocalService.cs:74]: Ok. Let's go ahead with this change.
Fingers crossed :)

**Review by Dor-bl — APPROVED** [review]

File: src/Appium.Net/Appium/Service/AppiumLocalService.cs (hunk 3 above)
Comment (nvborisenko) [reply src/Appium.Net/Appium/Service/AppiumLocalService.cs:74]: Selenium is already felt it.

**Comment by jlipps** [comment]
Hi @nvborisenko, congrats, the Appium project wants to compensate you for this contribution! Please reply to this comment mentioning me and sharing your [OpenCollective](https://opencollective.com) account name, so that we can initiate payment! Or let me know if you decline to receive compensation via OpenCollective. Thank you!
